replica/table.cc

```diff
@@ -3551,7 +3575,6 @@ future<> table::cleanup_compaction_groups(database& db, db::system_keyspace& sys
 
 future<> table::cleanup_tablet(database& db, db::system_keyspace& sys_ks, locator::tablet_id tid) {
     auto holder = async_gate().hold();
-    rwlock::holder exclusive_lock = co_await _sg_manager->get_rwlock().hold_write_lock();
     auto& sg = storage_group_for_id(tid.value());
```

Comment:
don't we need to enter the storage_group gate here?

Reply:
Otherwise, we should copy the sg compaction_groups  synchronously and pass that to `clear_inactive_reads_for_tablet` (we could do it there too, but I think it's cleaner to do it this way since the scope of usage is clearer)

Reply:
> Otherwise, we should copy the sg compaction_groups synchronously and pass that to `clear_inactive_reads_for_tablet` (we could do it there too, but I think it's cleaner to do it this way since the scope of usage is clearer)

I think it's not a problem, since cleanup_tablet() is the one deallocating the group (happens in the end).